### PR TITLE
GH-14761: [Dev] Update labels on PR labeler to use new Component ones

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Comment by raulcd** [comment]
* issue: #14761

File: .github/workflows/dev_pr/labeler.yml [hunk 1]
```diff
@@ -15,53 +15,53 @@
...
-lang-go:
+"Component: go":
   - go/**/*
 
-lang-java:
+"Component: java":
   - java/**/*
 
-lang-js:
+"Component: js":
```
Comment (jorisvandenbossche) [inline .github/workflows/dev_pr/labeler.yml:33]:
```suggestion
"Component: JavaScript":
```

On the short term, we should probably match the labels exactly as the components on JIRA (we can rename them after the issues migration)? So for example "js" is "JavaScript"

Comment (raulcd) [reply .github/workflows/dev_pr/labeler.yml]: Thanks! I have updated all the labels to match the existing ones on JIRA.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
And I renamed all relevant labels

**Pull request merged**